c2corg_ui/static/js/advancedsearch.js

```diff
@@ -0,0 +1,179 @@
+goog.require('app.Api');
+goog.require('ngeo.Location');
+goog.require('ol.Feature');
+goog.require('ol.format.GeoJSON');
+
+
+/**
+ * This directive is used to display the advanced search form and results.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.advancedSearchDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppAdvancedSearchController',
+    controllerAs: 'searchCtrl',
+    bindToController: true,
+    scope: true,
+    templateUrl: '/static/partials/advancedsearch.html'
+  };
+};
+
+
+app.module.directive('appAdvancedSearch', app.advancedSearchDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Directive scope.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {app.Api} appApi Api service.
+ * @param {ngeo.Location} ngeoLocation ngeo Location service.
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.AdvancedSearchController = function($scope, $attrs, appApi, ngeoLocation,
+    gettextCatalog) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.doctype_ = $attrs['documentType'];
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.resCounter_ = $attrs['resultsCounter'];
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {ngeo.Location}
+   * @private
+   */
+  this.location_ = ngeoLocation;
+
+  /**
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
+  /**
+   * @type {number}
+   * @export
+   */
+  this.total = 0;
+
+  /**
+   * @type {Array.<Object>}
+   * @export
+   */
+  this.documents = [];
+
+  // Get the initial results when loading the page.
+  // If a map is used, wait to get the map extent
+  // before triggering the request.
+  if (!$attrs['useMap']) {
+    this.getResults_();
+  }
+
+  // Refresh the results when pagination or criterias have changed:
+  this.scope_.$root.$on('searchFilterChange', this.getResults_.bind(this));
+};
+
+
+/**
+ * @private
+ */
+app.AdvancedSearchController.prototype.getResults_ = function() {
+  var url = this.location_.getUriString();
+  var qstr = goog.uri.utils.getQueryData(url) || '';
+  qstr += '&pl=' + this.gettextCatalog_.currentLanguage;
+  qstr = qstr.replace('debug', ''); // FIXME better handling of special params?
+  this.api_.listDocuments(this.doctype_, qstr).then(
+    this.successList_.bind(this)
+  );
+};
+
+
+/**
+ * @param {Object} response Response from the API server.
+ * @private
+ */
+app.AdvancedSearchController.prototype.successList_ = function(response) {
+  if (!('data' in response)) {
+    return;
+  }
+  this.documents = response['data']['documents'];
+  this.total = response['data']['total'];
+  if (this.resCounter_) {
+    $('#' + this.resCounter_).html(
+      /** @type {string} */ (this.total.toString()));
```

Comment:
It may looks odd indeed to cast the output of ``toString()`` to a string but I think I had a compiling error otherwise...